packages/engine/src/utils.rs

```diff
@@ -118,37 +123,50 @@ pub fn init_logger(std_err_output_format: OutputFormat, file_output_name: &str)
         ),
     };
 
-    let file_appender =
-        tracing_appender::rolling::never("./log", format!("{file_output_name}.log"));
-    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
+    let json_file_appender =
+        tracing_appender::rolling::never("./log", format!("{log_file_output_name}.log"));
+    let (non_blocking, _json_file_guard) = tracing_appender::non_blocking(json_file_appender);
 
     let json_file_layer = fmt::layer()
         .event_format(formatter.json())
         .fmt_fields(JsonFields::new())
         .with_writer(non_blocking);
 
+    let texray_file_appender =
+        tracing_appender::rolling::never("./log", format!("{texray_output_name}.txt"));
+    let (non_blocking, _tex_ray_guard) = tracing_appender::non_blocking(texray_file_appender);
+
+    // we clone update_settings to satisfy move rules as writer takes a `Fn` rather than `FnOnce`
+    let texray_layer =
```

Comment:
As per the PR description:
>  Do we want to include tracing-texray, it is Alpha. And we do not know the performance overhead. Perhaps an acceptable solution is to feature flag it

Reply:
I'd made `tracing-texray` an optional dependency disabled by default, maybe also enabled by a `debug` feature, which also enables spans at all. Most users wouldn't care about texray, so let's not hard depend on it.

Reply:
I think feature-flagging it, and using that to make it an optional dependency is probably the best strategy yep.

I don't think making it only there on debug builds is the right move, as texray is for timings and ideally we'll be doing that on release.

Reply:
Yep, my thought at the moment is to feature flag texray, and leave tracing and spans in by default. But in the future if we see it as a problem then we can even feature flag tracing.

Reply:
Sounds good to me!

Reply:
That's been done now, there's a feature called `texray` on the binaries and lib